frontend/packages/console-shared/src/components/namespace/NamespaceDropdown.tsx

```diff
@@ -0,0 +1,428 @@
+};
+
+/* ****************************************** */
+
+const NamespaceMenu: React.FC<{
+  setOpen: (isOpen: boolean) => void;
+  onSelect: (event: React.MouseEvent, itemId: string) => void;
+  selected?: string;
+  isProjects: boolean;
+  allNamespacesTitle: string;
+  onCreateNew: () => void;
+  menuRef: React.MutableRefObject<HTMLDivElement>;
+}> = ({ setOpen, onSelect, selected, isProjects, allNamespacesTitle, onCreateNew, menuRef }) => {
+  // const menuRef = React.useRef(null);
+  const filterRef = React.useRef(null);
+
+  const [filterText, setFilterText] = React.useState('');
+
+  // Bookmarking / favorites (note in <= 4.8 this feature was known as bookmarking)
+  const favoritesUserSettingsKey = `${NAMESPACE_USERSETTINGS_PREFIX}.bookmarks`;
+  const systemNamespacesSettingsKey = `${NAMESPACE_USERSETTINGS_PREFIX}.systemNamespace`;
+  const favoriteStorageKey = `${NAMESPACE_LOCAL_STORAGE_KEY}-bookmarks`;
+  const systemNamespaceKey = `${NAMESPACE_LOCAL_STORAGE_KEY}-systemNamespace`;
+  const [favorites, setFavorites] = useUserSettingsCompatibility(
+    favoritesUserSettingsKey,
+    favoriteStorageKey,
+    undefined,
+    true,
+  );
+
+  const canList: boolean = useFlag(FLAGS.CAN_LIST_NS);
+  const canCreate: boolean = useFlag(FLAGS.CAN_CREATE_PROJECT);
+  const [options, optionsLoaded] = useK8sWatchResource<K8sResourceKind[]>({
+    isList: true,
+    kind: isProjects ? ProjectModel.kind : NamespaceModel.kind,
+    optional: true,
+  });
+
+  const optionItems = React.useMemo(() => {
+    if (!optionsLoaded) {
+      return [];
+    }
+    const items = options.map((item) => {
+      const { name } = item.metadata;
+      return { title: name, key: name };
+    });
+    if (!items.some((option) => option.title === selected) && selected !== ALL_NAMESPACES_KEY) {
+      items.push({ title: selected, key: selected }); // Add current namespace if it isn't included
+    }
+    items.sort((a, b) => a.title.localeCompare(b.title));
+
+    if (canList) {
+      items.unshift({ title: allNamespacesTitle, key: ALL_NAMESPACES_KEY });
+    }
+    return items;
+  }, [allNamespacesTitle, canList, options, optionsLoaded, selected]);
+
+  const hasSystemNamespaces = React.useMemo(
+    () => optionItems.some((option) => isSystemNamespace(option)),
+    [optionItems],
+  );
+
+  const onSetFavorite = React.useCallback(
+    (key, active) => {
+      setFavorites((oldFavorites) => ({
+        ...oldFavorites,
+        [key]: active ? true : undefined,
+      }));
+    },
+    [setFavorites],
+  );
+
+  const [systemNamespaces, setSystemNamespaces] = useUserSettingsCompatibility(
+    systemNamespacesSettingsKey,
+    systemNamespaceKey,
+    false,
+    true,
+  );
+
+  const isFavorite = React.useCallback((option) => !!favorites?.[option.key], [favorites]);
+
+  const isOptionShown = React.useCallback(
+    (option, checkIsFavorite: boolean) => {
+      const containsFilterText = fuzzysearch(filterText.toLowerCase(), option.title.toLowerCase());
```

Comment:
I changed the filter matching to use fuzzy search. The current custom dropdown component uses fuzzy search for matching.

Line 285 was originally `const containsFilterText = option.title.toLowerCase().includes(filterText.toLowerCase());`